app/controllers/topics_controller.rb

```diff
@@ -43,6 +43,15 @@ def id_for_slug
     render json: { slug: topic.slug, topic_id: topic.id, url: topic.url }
   end
 
+  def show_by_external_id
+    if params[:external_id]
+      topic = Topic.find_by(external_id: params[:external_id])
```

Comment:
Do we need a guardian check here to see if they can see it?

Reply:
I don't think so because that is all done after the redirect in the `def show` method.

Reply:
but yea you might be right, i'll double check this.

Reply:
Probably better to just raise `Discourse::NotFound` right away so as not to reveal that the private topic actually exists. I think we do this for `Topics#show` already.

Reply:
https://github.com/discourse/discourse/blob/cd460155aa1ff77c8ff3b57799734eb23f37a983/app/controllers/topics_controller.rb#L65-L70